src/3d/Ast.fst

```diff
@@ -447,17 +448,23 @@ type struct_field = {
   field_action:option (action & bool); //boo indicates if the action depends on the field value
 }
 
-[@@ PpxDerivingYoJson ]
-let field = with_meta_t struct_field
+and atomic_field = with_meta_t atomic_field'
 
-[@@ PpxDerivingYoJson ]
-noeq
-type case =
+and field' =
+  | AtomicField of atomic_field
+  | RecordField : record -> ident -> field'
+  | SwitchCaseField : switch_case -> ident -> field'

```

Comment:
This is one of the main changes in the PR.
A field is no longer just an atomic field. It can be a composite field too, enabled nested structs (RecordField) and unions (SwitchCaseField)